From 5cbe8f759ab3d21ef1394fbe280f184e1c3c1b1a Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Tue, 23 Jun 2020 09:27:30 -0500
Subject: [PATCH 40/42] arm64: dts: beacon-imx8mm: Fix regulatory dependencies

Several of the drivers are expecting supply regulators and when they are
not specified, the drivers throw messages/warnings while booting.

These extra messages make our system look like there are more warnings,
so this patch assigns the missing regulators to clean up the boot messages.

Fixes: PHOENIX8-29

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 .../freescale/beacon-imx8mm-baseboard.dtsi    |  2 ++
 .../boot/dts/freescale/beacon-imx8mm-hdmi.dts | 23 +++++++++++++++++--
 .../boot/dts/freescale/beacon-imx8mm-som.dtsi | 18 +++++++++++++++
 3 files changed, 41 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
index bbe28512726a..d450a60cad51 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-baseboard.dtsi
@@ -426,6 +426,7 @@
 		reg = <0x20>;
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_pcal6414>;
+		vcc-supply = <&buck6_reg>;
 		gpio-controller;
 		#gpio-cells = <2>;
 		interrupt-parent = <&gpio4>;
@@ -435,6 +436,7 @@
 	pca6416_1: gpio@21 {
 		compatible = "nxp,pcal6416";
 		reg = <0x21>;
+		vcc-supply = <&buck6_reg>;
 		gpio-controller;
 		#gpio-cells = <2>;
 		interrupt-parent = <&gpio4>;
diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-hdmi.dts b/arch/arm64/boot/dts/freescale/beacon-imx8mm-hdmi.dts
index 270412e6a7bb..722e04c73072 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-hdmi.dts
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-hdmi.dts
@@ -17,7 +17,21 @@
 		stdout-path = &uart2;
 	};
 
-	reg_hdmi: regulator-lcd {
+	reg_hdmi_1p2: regulator-hdmi-avdd {
+		compatible = "regulator-fixed";
+		regulator-name = "hdmi_v1p2";
+		regulator-min-microvolt = <1200000>;
+		regulator-max-microvolt = <1200000>;
+	};
+
+	reg_hdmi_avdd: regulator-hdmi-avdd {
+		compatible = "regulator-fixed";
+		regulator-name = "hdmi_avdd";
+		regulator-min-microvolt = <1800000>;
+		regulator-max-microvolt = <1800000>;
+	};
+
+	reg_hdmi_dvdd: regulator-hdmi-dvdd {
 		pinctrl-names = "default";
 		pinctrl-0 = <&pinctrl_hdmi>;
 		compatible = "regulator-fixed";
@@ -34,7 +48,12 @@
 	adv_bridge: adv7535@3d {
 		compatible = "adi,adv7533";
 		reg = <0x3d>;
-		dvdd-supply = <&reg_hdmi>;
+		avdd-supply = <&reg_hdmi_avdd>;
+		a2vdd-supply = <&reg_hdmi_avdd>;
+		dvdd-supply = <&reg_hdmi_dvdd>;
+		pvdd-supply = <&reg_hdmi_avdd>;
+		v1p2-supply = <&reg_hdmi_1p2>;
+		v3p3-supply = <&reg_hdmi_dvdd>;
 		adi,addr-cec = <0x3b>;
 		adi,dsi-lanes = <4>;
 		pinctrl-names = "default";
diff --git a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
index 017f48d8e02e..7919347c693b 100644
--- a/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
+++ b/arch/arm64/boot/dts/freescale/beacon-imx8mm-som.dtsi
@@ -23,6 +23,7 @@
 	pinctrl-0 = <&pinctrl_fec1>;
 	phy-mode = "rgmii-id";
 	phy-handle = <&ethphy0>;
+	phy-supply = <&buck6_reg>;
 	fsl,magic-packet;
 	status = "okay";
 
@@ -223,6 +224,23 @@
 	};
 };
 
+&usbmisc1 {
+	vbus-wakeup-supply = <&buck6_reg>;
+};
+
+
+&usbmisc2 {
+	vbus-wakeup-supply = <&buck6_reg>;
+};
+
+&usbphynop1 {
+	vcc-supply = <&buck6_reg>;
+};
+
+&usbphynop2 {
+	vcc-supply = <&buck6_reg>;
+};
+
 &usdhc1 {
 	#address-cells = <1>;
 	#size-cells = <0>;
-- 
2.17.1

